depthai_nodes/parsing_neural_network.py

```diff
@@ -74,21 +61,25 @@ def build(
             object.
         """
 
-        if isinstance(nn_source, dai.NNModelDescription):
-            if not nn_source.platform:
-                nn_source.platform = (
-                    self.getParentPipeline().getDefaultDevice().getPlatformAsString()
+        if isinstance(slug, str):
+            if ":" not in slug:
+                raise ValueError(
+                    "Slug must be in the format <model_slug>:<model_version_slug>."
                 )
-            self._nn_archive = dai.NNArchive(dai.getModelFromZoo(nn_source))
-        elif isinstance(nn_source, dai.NNArchive):
-            self._nn_archive = nn_source
-        else:
-            raise ValueError(
-                "nn_source must be either a NNModelDescription or NNArchive"
+            model_slug, model_version_slug = slug.split(":")
```

Comment:
Have you tested this? Model slugs can now include ReferenceSlug like: `luxonis/yolov6-nano:r2-coco-512x288:b45ea75`

Reply:
We could evaluate this further but I just wanted to make sure that ":" is present so that next line doesn't fail. If the slug is incorrect, the code fails and the error is quite descriptive (`No model found for the given model description`) so I don't think additional checks are needed at this level.

Reply:
Yes but if two ":" are present in a string, the split will produce a list of 3 substrings, not the two defined and the code will throw an error even if valid slug is given.

Reply:
Thanks! I've added a fix in [5ff83dd](https://github.com/luxonis/depthai-nodes/pull/134/commits/5ff83ddf96cf68da4e4615f33aa2e6504bdb81ab).